media-libs/libgfx/libgfx-1.1.0.ebuild

```
@@ -10,9 +10,19 @@ SRC_URI="http://mgarland.org/dist/${P}.tar.gz"
 LICENSE="LGPL-2.1"
 SLOT="0"
 KEYWORDS="~amd64 ~x86"
-IUSE="static-libs"
+IUSE="static-libs test"
```

```suggestion
IUSE="static-libs test"
RESTRICT="!test? ( test )"
```
To enforce that USE="test" if FEATURES="test"

---

Actually, since there aren't any test dependencies or other things that are conditional on the test USE flag, you actually don't need to add the flag at all.

---

> 
> 
> Actually, since there aren't any test dependencies or other things that are conditional on the test USE flag, you actually don't need to add the flag at all.

I do not understand this comment. I removed test from IUSE in the second commit; however, now the tests are not performed. Is this desirable behavior? If so, that is what my second commit does.

---

There is USE="test" and FEATURES="test", and these do different things (the identical naming is slightly confusing). FEATURES="test" enables running the test phase when emerging an ebuild, so if I do `FEATURES="test" emerge foobar` emerge will execute the test phase of foobar after the compile phase and before the install phase.
USE="test" is a helper to pull in extra test dependencies or enable the building of tests if required. So my ebuild foobar might have:
```
DEPEND="test? ( some/test-dep )"

src_configure() {
       econf $(use_enable test)
}
```
if foobar happens to have a build system which optionally accepts `--enable-test` to build the test suite. In this case I have used the flag "test", so I must define `test` in IUSE. If I need a test flag, I should also add `RESTRICT="!test? ( test )"` (and this is where the identical naming gets confusing, the first "test" is the USE flag, the second is the FEATURES), this will tell emerge to only run the test phase if both USE="test" _and_ FEATURES="test".

Now in this case, there don't seem to be any test dependencies or anything else which is conditional on `test? ()`, so we don't need IUSE="test" or RESTRICT="!test? ( test )". Emerging the ebuild with FEATURES="test" _should_ enable running the test phase without the need to set any flag. The castxml can probably also lose the test flag, since it doesn't seem to be doing anything there.

---

Thank you for this explanation, I did not see it until now.